### PR TITLE
Added new API endpoints (logs, evidence)

### LINKED ISSUE #1528: [FR]: API endpoint to retrieve logs
### What is the feature you are proposing?

Create new API endpoints to retrieve log data

### What would this feature improve or what problem would it solve?

Create new API endpoints to retrieve log data

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by jleaniz**
### Description of the change

This PR creates new API endpoints to retrieve log data for Turbinia components in a similar way to the "tail" command.

It also adds a new endpoint to download evidence files based on evidence IDs.

### Applicable issues

- fixes #1528 

### Checklist

- [X] All tests were successful.
- [X] Unit tests added.
- [X] Documentation updated.

### CODE REVIEW

**Comment by jleaniz** [comment]
image

File: turbinia/api/utils.py [hunk 1]
```diff
@@ -189,3 +202,36 @@ async def create_tarball(output_path: str) -> AsyncGenerator[bytes, Any]:
 
     # Yield end-of-archive marker.
     yield stream.pop()
+
+
+def tail_log(log_path, max_lines=500) -> str:
```
Comment (hacktobeer) [inline turbinia/api/utils.py:207]: Very professional. Or you use subprocess and tail.....

File: turbinia/api/utils.py [hunk 2]
```diff
@@ -28,6 +28,19 @@
 
 log = logging.getLogger(__name__)
 
+ATTACHMENT_RESPONSE = {
```
Comment (hacktobeer) [inline turbinia/api/utils.py:31]: You can leave this out and specify it in the openapi.yaml file which is how we specify content-type for other repsonses as well (eg json).

```
responses:
        '200':
          content:
            application/octet-stream:
              schema:
                type: string
                format: binary
```

**Review by hacktobeer — COMMENTED** [review]
Nice one, love the log file endpoint with tailing.

File: turbinia/api/utils.py (hunk 1 above)
Comment (jleaniz) [reply turbinia/api/utils.py:207]: Got to love re-inventing the wheel :)

File: turbinia/api/routes/logs.py [hunk 3]
```diff
@@ -15,18 +15,62 @@
...
+
+  In some deployments, the same file can contain all logs (e.g. running all
+  services locally in the same container).
+  """
+  if not hostname:
+    return JSONResponse(content={'detail': 'Invalid hostname'}, status_code=404)
+
+  if 'NODE_NAME' in os.environ:
+    log_name = f'{hostname}.{os.environ["NODE_NAME"]!s}'
+  log_name = f'{hostname}.log'
```
[the remark this answers is not in the document]
Comment (jleaniz) [reply turbinia/api/routes/logs.py:70]: You're right about the overwrite, nice catch. 

On the NODE_NAME issue, yeah it's for k8s deployments. This is what the log filename is set in the logger module, it basically has the same check for NODE_NAME so I re-used this.

File: turbinia/api/openapi.yaml [hunk 4]
```diff
@@ -491,17 +521,91 @@ paths:
...
+        required: false
+        schema:
+          default: 500
+          exclusiveMinimum: 0.0
+          title: Num Lines
+          type: integer
+      responses:
+        '200':
+          content:
+            application/json:
```
Comment (hacktobeer) [inline turbinia/api/openapi.yaml:540]: is this an json response? or plain text like the config.download endpoint.
same for the other log endpoints.

File: turbinia/api/routes/download.py [hunk 5]
```diff
@@ -22,13 +22,15 @@
 from fastapi.requests import Request
 
 from turbinia import config as turbinia_config
+from turbinia.api import utils as api_utils
```
Comment (hacktobeer) [inline turbinia/api/routes/download.py:25]: if you don't use this import, remove it. If you do, I need new glasses ;)

File: turbinia/api/routes/evidence.py [hunk 6]
```diff
@@ -21,13 +21,14 @@
...
 from fastapi.requests import Request
-from fastapi.responses import JSONResponse
+from fastapi.responses import JSONResponse, FileResponse
 from typing import List, Annotated
 
 from turbinia import evidence
 from turbinia import config as turbinia_config
 from turbinia import state_manager
 from turbinia import TurbiniaException
+from turbinia.api import utils as api_utils
```
Comment (hacktobeer) [inline turbinia/api/routes/evidence.py:31]: used?

**Review by hacktobeer — COMMENTED** [review]
LGTM, feel free to merge after addresses small nit comments.

**Pull request merged**